src/lib/stitching/exchange/stitching.ts

```diff
@@ -248,6 +250,82 @@ export const exchangeStitchingEnvironment = ({
             })
           },
         },
+        artworkOrEditionSet: {
+          fragment: gql`
+            ... on CommerceLineItem {
+              artworkId
+              editionSetId
+            }
+          `,
+          resolve: async (parent, _args, context, info) => {
+            const artworkId = parent.artworkId
+            const editionSetId = parent.editionSetId
+
+            if (editionSetId) {
+              return info.mergeInfo.delegateToSchema({
+                schema: localSchema,
+                operation: "query",
+                fieldName: "artwork",
+                args: {
+                  id: artworkId,
+                },
+                context,
+                info,
+                transforms: [
+                  // Wrap document takes a subtree as an AST node
+                  new WrapQuery(
+                    // path at which to apply wrapping and extracting
+                    ["artwork"],
+                    (subtree: SelectionSetNode) => ({
```

Comment:
@sweir27 - curious if there was prior art here that you followed? Haven't seen this AST stuff in stitching before, its interesting.

Reply:
Mostly a lot of googling/docs reading, and [this helpful GitHub issue](https://github.com/apollographql/graphql-tools/issues/543) which was a jumping off point!